application/src/main/java/net/corda/v5/application/marshalling/MarshallingService.java

```diff
@@ -0,0 +1,54 @@
+package net.corda.v5.application.marshalling;
+
+import net.corda.v5.application.flows.ClientStartableFlow;
+import net.corda.v5.base.annotations.DoNotImplement;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * {@link MarshallingService} is an abstract interface for marshalling to and from formatted string data.
+ * Corda provides specialized implementations of the marshalling services for converting data in different string
+ * formats.
+ * <p>
+ * Example usage:
+ * @see ClientStartableFlow
+ */
+@DoNotImplement
+public interface MarshallingService {
+
+    /**
+     * Format the input data into the service's output format.
+     *
+     * @param data The object to convert on input.
+     *
+     * @return String representation of the data formatted according to the provided service.
+     */
+    @NotNull
+    String format(@NotNull Object data);
+
+    /**
+     * Parse input strings to strongly typed objects.
+     * <p>
+     * This method will throw an exception if the provided string does not conform to the expected format of the
+     * service.
+     *
+     * @param input The input string to parse.
+     * @param clazz The type to try and parse the data into.
+     *
+     * @return An instance of the required type containing the input data.
+     */
+    <T> T parse(@NotNull String input, @NotNull Class<T> clazz);
```

Comment:
Should we use `@NotNull` here too?

Reply:
Same reasoning as above. You are effectively asking for `<T : Any>` here - which we _might_ be able to do, but isn't what we had.

Reply:
Here it is advantageous to be nullable and return null for null